cfme/tests/cloud_infra_common/test_tag_visibility.py

```diff
@@ -25,3 +31,216 @@ def tagged_vm(tag, has_no_providers_modscope, setup_provider_modscope, provider)
 def test_tag_vis_vm(tagged_vm, user_restricted):
     with user_restricted:
         assert tagged_vm.exists, "vm not found"
+
+
+# For now use existing tags as for bz 1579867, should be replaced with random created tag fixture
+@pytest.fixture(scope='module')
+def location_tag():
+    """Existing tag object"""
+    category = Category(name='location',
+                        display_name='Location')
+    tag = Tag(name='paris',
+              display_name='Paris',
+              category=category)
+    return tag
+
+
+# For now use existing tags as for bz 1579867, should be replaced with random created tag fixture
+@pytest.fixture(scope='module')
+def service_level_tag():
+    """Existing tag object"""
+    category = Category(name='service_level',
+                        display_name='Service Level')
+    tag = Tag(name='silver',
+              display_name='Silver',
+              category=category)
+    return tag
+
+
+@pytest.fixture(scope='module')
+def vms_for_tagging(provider, request, appliance):
+    """Get two existing vms for tagging"""
+    view = navigate_to(provider, 'ProviderVms')
+    all_names = view.entities.all_entity_names
+    first_vm = appliance.collections.infra_vms.instantiate(name=all_names[0], provider=provider)
+    second_vm = appliance.collections.infra_vms.instantiate(name=all_names[1], provider=provider)
+    return first_vm, second_vm
+
+
+@pytest.fixture
+def group_with_tag_expression(appliance, user_restricted, request):
+    def _group_with_tag_expression(expression):
+        """Updates group with provided expression, also assign user to group"""
+        group = appliance.collections.groups.create(
+            description='grp{}'.format(fauxfactory.gen_alphanumeric()),
+            role='EvmRole-approver',
+            tag=expression
+        )
+        request.addfinalizer(group.delete)
+        user_restricted.update({'group': group})
+        return group
+
+    return _group_with_tag_expression
+
+
+@pytest.fixture
+def check_vm_visibility(user_restricted, appliance, request):
+    def _check_vm_visibility(group, vm, vis_expect):
+        """
+        Args:
+            group: restricted group with expression tag
+            vm: vm object to check visibility
+            vis_expect: bool, True if tag should be visible
+
+        Returns: None
+        """
+        with user_restricted:
+            view = navigate_to(appliance.server, 'LoggedIn')
+
+            orig_group = view.current_groupname
+            if group.description != orig_group:
+                view.change_group(group.description)
+            try:
+                navigate_to(vm, 'Details')
+                actual_visibility = True
+            except Exception:
```

Comment:
**REQUIRED**
Please be more specific, what kind of exception should raise here?